### PR TITLE
Debug travis builds with different node version

### PR DESCRIPTION
**Pull request by TheMarex**
# Issue

Travis currently fails to select the right node version.

## Tasklist
 - [x] review
 - [x] adjust for comments

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -36,6 +37,7 @@ env:
    - CCACHE_VERSION=3.3.1
    - CMAKE_VERSION=3.7.2
    - MASON="$(pwd)/scripts/mason.sh"
+   - NODE="4"
```
Comment (daniel-j-h) [inline .travis.yml]: I don't understand how this interacts with the `node_js` values set above. Is this just to have a default, and will the values above overwrite node 4 here?

File: .travis.yml [hunk 2]
```diff
@@ -230,6 +228,10 @@ matrix:
         - ./scripts/travis/publish.sh
 
 before_install:
+  - source $NVM_DIR/nvm.sh
+  - nvm install $NODE
+  - nvm use $NODE
+  - node --version
```
Comment (daniel-j-h) [inline .travis.yml]: nvm is installed on travis by default I assume?

File: .travis.yml (hunk 1 above)
Comment (TheMarex) [inline .travis.yml]: I need the values above to install `yarn` for me. I think it also causes node 4 and 6 to be in the nvm cache. That seems to work at least for non OSX builds which need an explicit `nvm install` step.

File: .travis.yml (hunk 2 above)
Comment (TheMarex) [inline .travis.yml]: Only if you set the global `node_js` though I think.

**Pull request merged**